src/mame/video/twin16.cpp

```diff
@@ -331,22 +331,22 @@ void twin16_state::draw_sprites( screen_device &screen, bitmap_ind16 &bitmap )
 				xpos = 320-xpos-width;
 				flipx = !flipx;
 			}
-			if( xpos>=320 ) xpos -= 65536;
-			if( ypos>=256 ) ypos -= 65536;
+			if( xpos>cliprect.max_x ) xpos -= 65536;
+			if( ypos>cliprect.max_y ) ypos -= 65536;
```

Comment:
This doesn’t seem correct at all – wrapping shouldn't happen at a different point due to partial updates.